src/ServerAgent.cpp

```diff
@@ -178,10 +180,10 @@ void HTTPServerAgent::ThreadMain()
 		}
 
 		if (resp.success) {
-			Json::Reader reader;
-			resp.success = reader.parse(resp.buffer, resp.data, false);
-			if (!resp.success)
-				resp.buffer = std::string("JSON parse error: " + reader.getFormattedErrorMessages());
+			//Json::Reader reader;
```

Comment:
@sturnclaw second is this section which I think deserialised a repsonse

Reply:
This will be `resp.data = Json::parse(resp.buffer); resp.success = !resp.data.is_null();`. Substitute the `!is_null()` check for an `is_object()` or `is_array()` depending on the wire format of your message type.

Reply:
You'll likely also need to wrap it in a try block or use a non-throwing method; still getting my development environment back up so that's an exercise left to the reader.